clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -570,6 +604,21 @@ public void requestUpdate() {
             this.metadata.requestUpdate();
         }
 
+        /*
+         * Keep track of any nodes we've ever seen. Add current
+         * alive nodes to this tracking list. 
+         * @param nodes Current alive nodes
+         */
+        private void updateNodesEverSeen(List<Node> nodes) {
+            for (Node n : nodes) {
+                if (!nodesEverSeenById.containsKey(n.id())) {
```

Comment:
Yeah, my point is that if broker is reprovisioned (i.e. it keeps the same brokerId, but moves to a new host or IP), we'll still have the old Node in the `nodesEverSeen` collection since we're only checking the id. It seems like we'd want a check to see if something has changed and update. Maybe I'm missing something?